recon/src/recon/tests.rs

```diff
@@ -544,7 +545,7 @@ async fn response_is_synchronized() {
     let response = a.process_messages(&response.messages).await.unwrap();
     assert!(!response.is_synchronized);
     let response = x.process_messages(&response.messages).await.unwrap();
-    assert!(!response.is_synchronized);
+    assert!(response.is_synchronized);
```

Comment:
Why did this test invert?

Reply:
Can you update the test comments if this is expected?

Reply:
The `is_synchronized` logic changed so that it can detect synchronization one round earlier. Now, we don't have to wait for a round where nothing is exchanged so detect that we're done. We're synchronized as soon as the ranges match.